lib/auth/init.go

```diff
@@ -502,18 +503,23 @@ func initCluster(ctx context.Context, cfg InitConfig, asrv *Server) error {
 	}
 	span.AddEvent("completed migration legacy resources")
 
-	span.AddEvent("creating preset roles")
 	// Create presets - convenience and example resources.
-	if err := createPresetRoles(ctx, asrv); err != nil {
-		return trace.Wrap(err)
-	}
-	span.AddEvent("completed creating preset roles")
+	isDashboard := !modules.GetModules().Features().Cloud && modules.GetModules().Features().RecoveryCodes
+	if !isDashboard {
+		span.AddEvent("creating preset roles")
+		if err := createPresetRoles(ctx, asrv); err != nil {
+			return trace.Wrap(err)
+		}
+		span.AddEvent("completed creating preset roles")
 
-	span.AddEvent("creating preset users")
-	if err := createPresetUsers(ctx, asrv); err != nil {
-		return trace.Wrap(err)
+		span.AddEvent("creating preset users")
+		if err := createPresetUsers(ctx, asrv); err != nil {
+			return trace.Wrap(err)
+		}
+		span.AddEvent("completed creating preset users")
+	} else {
+		span.AddEvent("skipping preset role and user creation")
```

Comment:
I would just log this. You don't need to put it in the span.

Reply:
Switched to logging this here: [1054acc](https://github.com/gravitational/teleport/pull/37610/commits/1054acc35e06ef535174f1eb7923c57d20805686).